android/src/main/java/com/it_nomads/fluttersecurestorage/ciphers/StorageCipher18Implementation.java

```diff
@@ -33,22 +34,28 @@ public StorageCipher18Implementation(Context context) throws Exception {
         SharedPreferences.Editor editor = preferences.edit();
 
         String aesKey = preferences.getString(AES_PREFERENCES_KEY, null);
-        byte[] key;
-
-        if (aesKey == null) {
-            key = new byte[keySize];
-            secureRandom.nextBytes(key);
-            secretKey = new SecretKeySpec(key, KEY_ALGORITHM);
-
-            byte[] encryptedKey = rsaCipher.wrap(secretKey);
-            editor.putString(AES_PREFERENCES_KEY, Base64.encodeToString(encryptedKey, Base64.DEFAULT));
-            editor.commit();
-        } else {
-            byte[] encrypted = Base64.decode(aesKey, Base64.DEFAULT);
-            secretKey = rsaCipher.unwrap(encrypted, KEY_ALGORITHM);
-        }
 
         cipher = Cipher.getInstance("AES/CBC/PKCS7Padding");
+
+        if (aesKey != null) {
+            byte[] encrypted;
+            try {
+                encrypted = Base64.decode(aesKey, Base64.DEFAULT);
+                secretKey = rsaCipher.unwrap(encrypted, KEY_ALGORITHM);
+                return;
+            } catch (Exception e) {
+                Log.e("StorageCipher18Impl", "unwrap key failed", e);
+                encrypted = new byte[0];
+            }
+        }
+
+        byte[] key = new byte[keySize];
+        secureRandom.nextBytes(key);
+        secretKey = new SecretKeySpec(key, KEY_ALGORITHM);
+
+        byte[] encryptedKey = rsaCipher.wrap(secretKey);
+        editor.putString(AES_PREFERENCES_KEY, Base64.encodeToString(encryptedKey, Base64.DEFAULT));
```

Comment:
It wouldn't be possible to decrypt old data, that was saved with previous (broken) key.

Reply:
It's true. But I don't know why this bug only happens when running on release mode. In release mode, although I uninstall the previous app and reinstall it, I still get a previous key from SharePreference and this key is broken. Very strange. And when I run it on another device, it works without this problem. So, my code temporarily gets around this problem by deleting that 'ghost' key and create a new.

Reply:
Just curious, 
could you try https://github.com/mogol/flutter_secure_storage/issues/13#issuecomment-421083742 this? would it help you?